Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py

```diff
@@ -271,6 +286,7 @@ def _on_timeout(_):
         self.deferred.addErrback(_on_add_failed)
         return NOT_DONE_YET
 
+    @db_session
```

Comment:
Oh that's good to know! I wonder why they conflict?

Reply:
Because Pony does not know what to do with the `db_session` lock if it encounters a `yield` statement. It raises an error. Basically, there can be no `yields` inside the lock region.